src/rpc/handlers/AccountObjects.cpp

```diff
@@ -56,49 +56,50 @@ doAccountNFTs(Context const& context)
     if (auto const status = getLimit(request, limit); status)
         return status;
 
-    ripple::uint256 marker;
-    if (auto const status = getHexMarker(request, marker); status)
+    ripple::uint256 pageMarker;
+    ripple::uint256 tokenMarker;
+    if (auto const status =
+            getAccountNFTMarker(request, pageMarker, tokenMarker);
+        status)
         return status;
+    ripple::uint256 const maskedMarker = tokenMarker & ripple::nft::pageMask;
 
-    auto const first =
-        ripple::keylet::nftpage(ripple::keylet::nftpage_min(accountID), marker);
-    auto const last = ripple::keylet::nftpage_max(accountID);
-
-    auto const key =
-        context.backend
-            ->fetchSuccessorKey(first.key, lgrInfo.seq, context.yield)
-            .value_or(last.key);
-    auto const blob = context.backend->fetchLedgerObject(
-        ripple::Keylet(ripple::ltNFTOKEN_PAGE, key).key,
-        lgrInfo.seq,
-        context.yield);
-
-    std::optional<ripple::SLE const> cp{
-        ripple::SLE{ripple::SerialIter{blob->data(), blob->size()}, key}};
+    response[JS(account)] = ripple::toBase58(accountID);
 
     std::uint32_t cnt = 0;
     response[JS(account_nfts)] = boost::json::value(boost::json::array_kind);
     auto& nfts = response.at(JS(account_nfts)).as_array();
 
-    bool pastMarker = marker.isZero();
-    ripple::uint256 const maskedMarker = marker & ripple::nft::pageMask;
+    // start iteration at the last page, which is guaranteed to exist if the
+    // account owns any NFTs
+    auto const last = ripple::keylet::nftpage_max(accountID);
+    bool hasMarker = !pageMarker.isZero();
+    auto const key = hasMarker ? pageMarker : last.key;
```

Comment:
nit:

```C++
auto const key = hasMarker ? pageMarker : ripple::keylet::nftpage_max(accountID);
```

just to get rid of one line and not take up stack memory space for `last`